staging/src/k8s.io/kubectl/pkg/generate/versioned/service_basic.go

```diff
@@ -86,15 +87,11 @@ func (ServiceExternalNameGeneratorV1) ParamNames() []generate.GeneratorParam {
 func parsePorts(portString string) (int32, intstr.IntOrString, error) {
 	portStringSlice := strings.Split(portString, ":")
 
-	port, err := strconv.Atoi(portStringSlice[0])
+	port, err := utilsnet.ParsePort(portStringSlice[0], true)
 	if err != nil {
 		return 0, intstr.FromInt(0), err
 	}
 
-	if errs := validation.IsValidPortNum(port); len(errs) != 0 {
-		return 0, intstr.FromInt(0), fmt.Errorf(strings.Join(errs, ","))
-	}
-
```

Comment:
why removing the above validation check?

Reply:
`utilsnet.ParsePort` has defined the bitSize of the port in `portInt, err := strconv.ParseUint(port, 10, 16)`, so validating the range of the port is duplicated.
https://github.com/kubernetes/utils/blob/4a6ff033650d9ce34359cf57e8bc1e64f4a25146/net/net.go#L140-L146